src/main/kotlin/com/bridgecrew/ui/buttons/SuppressionButton.kt

```diff
@@ -33,7 +35,16 @@ class SuppressionButton(private var result: BaseCheckovResult): CheckovLinkButto
             Messages.showInfoMessage("File type $fileType cannot be suppressed inline", "Prisma Cloud");
             return
         }
-        val suppressionComment = generateCheckovSuppressionComment()
+
+        val dialog = SuppressionDialog()
+        dialog.show()
+        if(dialog.exitCode == DialogWrapper.OK_EXIT_CODE) {
+            generateComment(fileType, dialog.userJustification)
+        }
+    }
+
+    private fun generateComment(fileType: FileType, userReason: String?) {
+        val suppressionComment = generateCheckovSuppressionComment(userReason)
```

Comment:
doesn't that needs to be according to the file type?

Reply:
It does, i will add support for other types when they will be supported in `generateCheckovSuppressionComment` 
For the rest, the string itself is the same across different types (for now) except K8s.
Better handle it in another task IMO (we need to add a check if the yaml is a k8s file by some attributes)